src/zenml/config/compiler.py

```diff
@@ -293,10 +327,13 @@ def _get_step_spec(self, step: "BaseStep") -> StepSpec:
         Returns:
             The step spec.
         """
+        assert step.pipeline_parameter_name
```

Comment:
This is actually just to silence mypy, this will always be set if the step is part of a pipeline. However you mentioning that just made me catch some potential bug when adding a step to two pipelines which would overwrite this. I'll include a fix for that in this PR as well.

Reply:
I actually completely removed the parameter, this wasn't used anywhere outside of the compiler anyway where we have other means of getting this value